src/contract_client/client.ts

```diff
@@ -45,6 +47,45 @@ export class ContractClient {
     });
   }
 
+  /**
+   * Generate a ContractClient instance from the ContractClientOptions and the wasm binary
+   */
+  static async fromWasm(options: ContractClientOptions, wasm: BufferSource): Promise<ContractClient> {
+    const wasmModule = await WebAssembly.compile(wasm);
+    const xdrSections = WebAssembly.Module.customSections(wasmModule, "contractspecv0");
+    if (xdrSections.length === 0) {
+      return Promise.reject(new Error('Could not obtain contract spec from wasm'));
+    }
+    const section = xdrSections[0];
+    const bufferSection = Buffer.from(section);
+    const specEntryArray = processSpecEntryStream(bufferSection);
+    const spec = new ContractSpec(specEntryArray);
+    return new ContractClient(spec, options);
+  }
+
+  /**
+   * Generate a ContractClient instance from the contractId and rpcUrl
+   */
+  static async from(options: ContractClientOptions): Promise<ContractClient> {
+    if (!options || !options.rpcUrl || !options.contractId) {
+      return Promise.reject(new TypeError('options must contain rpcUrl and contractId'));
+    }
+    const { rpcUrl, contractId, allowHttp } = options;
+    const serverOpts: Server.Options = { allowHttp: allowHttp?? !rpcUrl.startsWith('https') };
+    const server = new Server(rpcUrl, serverOpts);
+    const contractLedgerKey = new Contract(contractId).getFootprint();
+    const response = await server.getLedgerEntries(contractLedgerKey);
```

Comment:
Perhaps the contract --> wasm lookup logic could be a method offered by `server`. Seems useful outside of this usecase.

Reply:
Agreed. It's [documented](https://developers.stellar.org/network/soroban-rpc/api-reference/methods/getLedgerEntries#requesting-a-contracts-wasm-code) but a built-in would be better; I opened https://github.com/stellar/js-stellar-sdk/issues/961.

Reply:
added, will add unit tests as well